src/pyspiffe/svid/jwt_svid_validator.py

```diff
@@ -77,38 +78,36 @@ def validate_claims(
         """Validates payload for required claims (aud, exp, sub).
 
         Args:
-            payload: token playload.
-            expected_audience: audience as a list of strings used to validate the 'aud' claim.
+            payload: Token payload.
+            expected_audience: Audience as a list of strings used to validate the 'aud' claim.
 
         Returns:
             None
 
         Raises:
-            InvalidClaimError: in case a required claim is not present in payload or expected_audience is not a subset of audience_claim.
-            TokenExpiredError: in case token is expired.
-            ValueError: in case expected_audience is empty.
+            MissingClaimError: In case a required claim is not present.
+            InvalidClaimError: In case a claim contains an invalid value or expected_audience is not a subset of audience_claim.
+            TokenExpiredError: In case token is expired.
+            ValueError: In case expected_audience is empty.
         """
         for claim in self._REQUIRED_CLAIMS:
             if not payload.get(claim):
-                raise InvalidClaimError(claim)
-        aud = payload.get('aud')
-        if aud is not None:
-            self._validate_exp(str(payload.get('exp')))
-            self._validate_aud(aud, expected_audience)
+                raise MissingClaimError(claim)
+
+        self._validate_exp(str(payload.get('exp')))
+        self._validate_aud(payload.get('aud', []), expected_audience)
```

Comment:
Here is where I changed the exception to make a difference between missing a required claim vs an invalid value.

Besides that, `aud` is part of the `_REQUIRED_CLAIMS` so `if aud is not None:` makes no sense. I removed it. Please, let me know if I'm missing something here.